test/functional/p2p_blocksonly.py

```diff
@@ -78,6 +79,34 @@ def blocksonly_mode_tests(self):
         self.nodes[0].disconnect_p2ps()
         self.generate(self.nodes[0], 1)
 
+    # Tests with an outbound peer that sets txrelay to false (which they do if they are in blocksonly mode)
+    def blocksonly_peer_tests(self):
+        self.restart_node(0, ["-noblocksonly"])  # disables blocks only mode
+        self.log.info("Check that we sustain full-relay outbound connections to blocksonly peers when not at the full outbound limit")
+        blocksonly_peer = self.nodes[0].add_outbound_p2p_connection(P2PInterface(), p2p_idx=1, txrelay=False)
+        # Trigger CheckForStaleTipAndEvictPeers, which is scheduled every 45s (EXTRA_PEER_CHECK_INTERVAL)
+        self.nodes[0].mockscheduler(46)
+        blocksonly_peer.sync_with_ping()
+        blocksonly_peer.peer_disconnect()
+
+        self.log.info("Check that we evict full-relay outbound connections to blocksonly peers at the full outbound limit")
```

Comment:
68769ff7c3252c7cfb8758459218979eacba7acb

You could expand this test by adding some irrelevant peers (e.g. ADDRFETCH), but no big deal.

Reply:
I tried it, but it turned out to be annoying because in the test we simulate being at the full outbound limit by artificially running with a lower `-maxconnections`. If we now also add an unrelated peer like ADDRFETCH , it would make the test fail because we don't have enough slots  in `semOutbound`.